src/main/scala/za/co/absa/rapidgen/swagger/SwaggerConfig.scala

```diff
@@ -16,28 +16,63 @@
 package za.co.absa.rapidgen.swagger
 
 import com.fasterxml.classmate.TypeResolver
+import org.slf4s.Logging
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.ApplicationContext
 import org.springframework.context.annotation.{Bean, Configuration}
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+import springfox.documentation.PathProvider
 import springfox.documentation.builders.{PathSelectors, RequestHandlerSelectors}
 import springfox.documentation.spi.DocumentationType
 import springfox.documentation.spring.web.plugins.Docket
 import springfox.documentation.swagger2.annotations.EnableSwagger2
+import za.co.absa.rapidgen.{Constants, RapidGenConfig}
 
 @Configuration
 @EnableSwagger2
-class SwaggerConfig @Autowired()(val typeResolver: TypeResolver)
+class SwaggerConfig(@Autowired() val typeResolver: TypeResolver,
+                    @Autowired val appContext: ApplicationContext)
   extends WebMvcConfigurer
     with SwaggerScalaTypesRules
     with SwaggerUISupport {
 
-  @Bean def api: Docket =
-    new Docket(DocumentationType.SWAGGER_2).
-      forCodeGeneration(true).
+  @Bean def api: Docket = {
+    var docket = new Docket(DocumentationType.SWAGGER_2).
+      forCodeGeneration(true)
+
+    if (appContext.isInstanceOf[SwaggerDocGenAppContext]) {
+      val rapidGenConfig = appContext.asInstanceOf[SwaggerDocGenAppContext].getRapidGenConfig
+      if (rapidGenConfig.isDefined) {
+        if (rapidGenConfig.get.overrideHost.isDefined) {
+          docket = docket.host(blankReplace(rapidGenConfig.get.overrideHost.get,
+            Constants.BLANK_HOST_PLACE_HOLDER))
+        }
+        if (rapidGenConfig.get.overrideBasePath.isDefined) {
+          val basePath = blankReplace(rapidGenConfig.get.overrideBasePath.get,
+            Constants.BLANK_BASE_PATH_PLACE_HOLDER)
+          docket = docket.pathProvider(new PathProvider() {
+            override def getApplicationBasePath: String = basePath
+
+            override def getOperationPath(operationPath: String): String = operationPath.replace(basePath, "")
+
+            override def getResourceListingPath(groupName: String, apiDeclaration: String): String = null
+          })
+        }
+      }
+    }
+    docket.
       select.
       apis(RequestHandlerSelectors.any).
       paths(PathSelectors.any).
       build
+  }
 
   @Bean def rpbPlugin = new SwaggerRequiredPropertyBuilderPlugin
+
+  private def blankReplace(original: String, replacement: String): String = {
+    if (original != null && !original.isBlank)
+      original
+    else
+      replacement
```

Comment:
```suggestion
    if (host.isBlank) Constants.BLANK_HOST_PLACE_HOLDER else host
```
Null checking seems redundant here. The method is private, has only one usage and by judging from the caller (option binding) the `host` parameter can logically never be null. `Some(null)` can only happen erroneously and should be captured on other levels. Normally we can assume that `Option` is null safe a priory. You should not guard from `Some(null)` as otherwise you'd need to guard pretty much every option binding which would bloat code up with those ifs.